drivers/serial/uart_gecko.c

```diff
@@ -18,18 +19,75 @@
 #define CLOCK_USART(id) CLOCK_ID_PRFX(USART_PREFIX, id)
 #define CLOCK_UART(id) CLOCK_ID_PRFX(UART_PREFIX, id)
 
+/* Helper define to determine if SOC supports hardware flow control */
+#if ((_SILICON_LABS_32B_SERIES > 0) || \
+	(defined(_USART_ROUTEPEN_RTSPEN_MASK) && \
+	defined(_USART_ROUTEPEN_CTSPEN_MASK)))
+#define HW_FLOWCONTROL_IS_SUPPORTED_BY_SOC
+#endif
+
+#define DT_DRV_COMPAT silabs_gecko_uart
+
+#if (DT_INST_PROP(0, hw_flow_control) || \
+	DT_INST_PROP(1, hw_flow_control))
+#define UART_GECKO_UART_HW_FLOW_CONTROL_ENABLED
+#endif
+
+#undef DT_DRV_COMPAT
+#define DT_DRV_COMPAT silabs_gecko_usart
+
+#if (DT_INST_PROP(0, hw_flow_control) ||    \
+	DT_INST_PROP(1, hw_flow_control) || \
+	DT_INST_PROP(2, hw_flow_control) || \
+	DT_INST_PROP(3, hw_flow_control) || \
+	DT_INST_PROP(4, hw_flow_control) || \
+	DT_INST_PROP(5, hw_flow_control))
```

Comment:
@galak is there a macro we can use to detect if this property exists on any instance? I worry that this will break if there is ever a 6th instance

Reply:
We dont have a macro right now.  Should see if this can be written using `DT_INST_FOREACH_STATUS_OKAY()`

@mbolivar-nordic any thoughts here?

Reply:
> @mbolivar-nordic any thoughts here?

@galak @MaureenHelm  yep, you can use `DT_INST_FOREACH_STATUS_OKAY()` for this as long as you're actually asking the question about status okay instances. The test as written is about any instance at all, not just status okay instances, so it's a bit different. (That said, I suspect the code as written might be wrong since I'm guessing disabled instances should not be taken into account, but that's just a hunch.)

The rest of this comment applies to the "status okay" case, but it would be trivial to add `DT_INST_FOREACH()` to devicetree.h if you really need the result for *all* instances, including disabled ones.

Any question "is `MY_TEST(inst)` true for any status okay `DT_DRV_COMPAT` instance number `inst`?" can be written as something like a [fold](https://en.wikipedia.org/wiki/Fold_%28higher-order_function%29) using DT_INST_FOREACH_STATUS_OKAY.

A code template is:

```
#define MY_FOLD_OP(inst) MY_TEST(inst) ||
#define MY_FOLD_RESULT   DT_INST_FOREACH_STATUS_OKAY(MY_FOLD_OP) 0
```

When expanded:

```
MY_FOLD_RESULT ->
MY_FOLD_OP(0) MY_FOLD_OP(1) ... MY_FOLD_OP(n-1) 0 ->
MY_TEST(0) || MY_TEST(1) || ... MY_TEST(n-1) || 0
```

So as long as `MY_TEST` expands to an expression the preprocessor can evaluate, the entire `MY_FOLD_RESULT` does too.

In this case:

```
#define HAS_HFC_OR(inst) DT_INST_NODE_HAS_PROP(inst, hw_flow_control) ||
#define ANY_INST_HAS_HFC DT_INST_FOREACH_STATUS_OKAY(HAS_HFC_OR) 0

#if ANY_INST_HAS_HFC

...

#endif
```

@galak I will leave it up to you to decide if we should add something like `DT_ANY_INST_STATUS_OKAY(MY_TEST)` to devicetree.h based on the above.

Reply:
Thank you, that is a good improvement. I included your suggestion.